src/mscorlib/src/System/Double.cs

```diff
@@ -44,56 +44,79 @@ public struct Double : IComparable, IFormattable, IConvertible
 
         internal static double NegativeZero = BitConverter.Int64BitsToDouble(unchecked((long)0x8000000000000000));
 
+        /// <summary>Determines whether the specified value is finite (zero, subnormal, or normal).</summary>
+        [Pure]
+        [System.Runtime.Versioning.NonVersionable]
+        public unsafe static bool IsFinite(double d)
+        {
+            var bits = BitConverter.DoubleToInt64Bits(d);
+            return (bits & 0x7FFFFFFFFFFFFFFF) < 0x7FF0000000000000;
+        }
+
+        /// <summary>Determines whether the specified value is infinite.</summary>
         [Pure]
         [System.Runtime.Versioning.NonVersionable]
         public unsafe static bool IsInfinity(double d)
         {
-            return (*(long*)(&d) & 0x7FFFFFFFFFFFFFFF) == 0x7FF0000000000000;
```

Comment:
`var bits = BitConverter.DoubleToInt64Bits(d);` is significantly more readable and gets inlined to the same code (marking with `AgressiveInlining` would probably be beneficial though).